pkg/ipam/types.go

```diff
@@ -114,3 +115,19 @@ func (ipam *IPAM) DebugStatus() string {
 	ipam.allocatorMutex.RUnlock()
 	return str
 }
+
+// GetVpcCIDRs returns all the CIDRs associated with the VPC this node belongs to.
+// This works only cloud provider IPAM modes and returns nil for other modes.
+func (ipam *IPAM) GetVpcCIDRs() (vpcCIDRs []*cidr.CIDR) {
+	sharedNodeStore.mutex.RLock()
+	defer sharedNodeStore.mutex.RUnlock()
+	primary, secondary := deriveVpcCIDRs(sharedNodeStore.ownNode)
+	if primary == nil {
+		return nil
+	}
+	if secondary == nil {
+		vpcCIDRs = make([]*cidr.CIDR, 1)
+		return append(vpcCIDRs, primary)
+	}
+	return append(secondary, primary)
```

Comment:
I think this code can be simplified

```suggestion
	primary, secondary := deriveVpcCIDRs(sharedNodeStore.ownNode)
	return append(primary, secondary)
```

Reply:
Was trying to avoid having nil items in the slice. Modified the secondary check to make it a little more clean `return []*cidr.CIDR{primary}`